src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,152 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private int nodeIndex;
```

Comment:
Totally not undestand why we need this class variable.
Mostly it's just create additional code in methods.
Instead of checking (nodeIndex == 0) or (size - nodeIndex != 1) you may just compare (currentNode == head) or  (currentNode != tail) etc. As in this case we check if it's the same object
```suggestion
```